service/src/main/scala/za/co/absa/loginsvc/rest/service/JWTService.scala

```diff
@@ -82,6 +87,40 @@ class JWTService @Autowired()(jwtConfigProvider: JwtConfigProvider) {
     val jwk = rsaPublicKey
     new JWKSet(jwk).toPublicJWKSet
   }
+
+  private def refreshSecrets(): Unit = {
+      val scheduler = Executors.newSingleThreadScheduledExecutor()
+      val scheduledFuture = scheduler.scheduleAtFixedRate(() => {
+        logger.info("Refreshing Keys")
+        try keyPair = jwtConfig.keyPair
+        catch {
```

Comment:
So for the in-memory implementation it doesn't always return the same result, a new key is generated on every refresh. I did it this way so that the user may implement key rotation with the in-memory option.

I do see a point of what if you don't want that to happen though, so I have set the config for key rotation and key polling to an option so that it's optional if the user wants to implement and scheduling may be turned off.

Reply:
I see, expected the old behavior and missed this update. This is cool 😎